### PR TITLE
Remove unnecessary Loc bounds check

### PR DESCRIPTION
**Pull request by elliottt**
We shouldn't expect the loc to be correct for internal methods used for metadata storage, in the rbi generator.

### Motivation

Fixing a TODO introduced in #5067 .

### Test plan

See included automated tests.

### CODE REVIEW

File: core/Loc.cc [hunk 1]
```diff
@@ -257,12 +257,7 @@ optional<string_view> Loc::source(const GlobalState &gs) const {
         return nullopt;
     } else {
         auto source = this->file().data(gs).source();
-        // TODO: this should be unnecessary
-        if (endPos() <= source.size()) {
-            return source.substr(beginPos(), endPos() - beginPos());
-        } else {
-            return nullopt;
-        }
+        return source.substr(beginPos(), endPos() - beginPos());
```
Comment (jez) [inline core/Loc.cc:260]: 🎉

**Review by jez — APPROVED** [review]
Is there an rbi-gen test we can write to prevent regressions?

**Comment by elliottt** [comment]
> Is there an rbi-gen test we can write to prevent regressions?

I've been trying to come up with a reproducer for the crash, but it's proving difficult. I'll try a bit more and merge this without if I can't get one.

**Comment by elliottt** [comment]
> > Is there an rbi-gen test we can write to prevent regressions?
> 
> I've been trying to come up with a reproducer for the crash, but it's proving difficult. I'll try a bit more and merge this without if I can't get one.

The real crash seems to be a resolver bug that we're triggering: the location generated for the mixes_in_class_methods fake method is incorrect.

**Pull request merged**